main.go

```diff
@@ -61,6 +61,7 @@ type HTTPProbe struct {
 	TLSConfig              config.TLSConfig  `yaml:"tls_config"`
 	Protocol               string            `yaml:"protocol"`              // Defaults to "tcp".
 	PreferredIpProtocol    string            `yaml:"preferred_ip_protocol"` // Defaults to "ip6".
+	DisableGzipEncoding    bool              `yaml:"disable_gzip_encoding"`
```

Comment:
This doesn't feel like a clean API, as we're leaking implementation details of the Go http api. Maybe we should just disable this for everyone?